File: src/extras-kconfigwidgets/libkhelpclient.cpp
#include <KHelpClient>
#include <QString>
#include <khelpclient.h>
#include "libkhelpclient.h"
#include "libkhelpclient.hxx"

void KHelpClient_InvokeHelp(const libqt_string anchor, const libqt_string appname) {
    QString anchor_QString = QString::fromUtf8(anchor.data, anchor.len);
    QString appname_QString = QString::fromUtf8(appname.data, appname.len);
    KHelpClient::invokeHelp(anchor_QString, appname_QString);
}

File: include/extras-kconfigwidgets/libkhelpclient.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");

/// ### [Upstream resources](https://api.kde.org/khelpclient.html)
pub const KHelpClient = extern struct {
    /// ### [Upstream resources](https://api.kde.org/khelpclient.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.KHelpClient,

    pub const _is_KHelpClient = {};

    /// ### [Upstream resources](https://api.kde.org/khelpclient.html#invokeHelp)
    ///
    /// ## Parameter(s):
    ///
    /// ` anchor: []const u8 `
    ///
    /// ` appname: []const u8 `
    ///
    pub fn InvokeHelp(anchor: []const u8, appname: []const u8) void {
        const anchor_str = qtc.libqt_string{
            .len = anchor.len,
            .data = anchor.ptr,
        };
        const appname_str = qtc.libqt_string{
            .len = appname.len,
            .data = appname.ptr,
        };
        qtc.KHelpClient_InvokeHelp(anchor_str, appname_str);
    }
};
